h2/src/main/org/h2/mvstore/db/MVPlainTempResult.java

```diff
@@ -22,21 +22,25 @@
  */
 class MVPlainTempResult extends MVTempResult {
 
+    private final ValueDataType valueType;
     private final MVMap<ValueLong, ValueArray> map;
     private long counter;
 
+    private MVMap<ValueArray, Boolean> index;
```

Comment:
If we have just one MVMap<ValueArray, Long>, then we can implement contains() without needing an index?

Reply:
But this will force sorting of results.

Reply:
As an optimization we can notify `LocalResult` whether it will be used in normal way or for `contains()` operations. In the first case we don't need this index, in the second case we can create a `MVStortedTempResult` instead. But this is a separate problem.